src/Microsoft.AspNet.Razor/Tokenizer/CSharpKeywordDetector.cs

```diff
@@ -88,7 +88,8 @@ internal static class CSharpKeywordDetector
             { "interface", CSharpKeyword.Interface },
             { "break", CSharpKeyword.Break },
             { "checked", CSharpKeyword.Checked },
-            { "namespace", CSharpKeyword.Namespace }
+            { "namespace", CSharpKeyword.Namespace },
+            { "when", CSharpKeyword.When }
```

Comment:
I'd rather stick with consistency here. If we want to break out keywords from this list lets do it in a bulk push instead of piecemealed.